src/test/java/co/rsk/federate/signing/hsm/message/AdvanceBlockchainMessageTest.java

```diff
@@ -82,4 +84,99 @@ public void test_getParsedBrothers_invalid_blockHeader() throws HSMBlockchainBoo
 
         message.getParsedBrothers(Hex.toHexString(invalidBlockHeader.getFullEncoded()));
     }
+
+    @Test
+    public void test_getParsedBrothers_sorted_by_hash() throws HSMBlockchainBookkeepingRelatedException {
+        BlockHeader block1Header = blockHeaderBuilder.setNumber(1).build();
+        BlockHeader block2Header = blockHeaderBuilder.setNumber(2).build();
+
+        List<BlockHeader> block1Brothers = Arrays.asList(
+            TestUtils.createBlockHeaderMock(102),
+            TestUtils.createBlockHeaderMock(105),
+            TestUtils.createBlockHeaderMock(103),
+            TestUtils.createBlockHeaderMock(101),
+            TestUtils.createBlockHeaderMock(104)
+        );
+        List<BlockHeader> block2Brothers = Arrays.asList(
+            TestUtils.createBlockHeaderMock(302),
+            TestUtils.createBlockHeaderMock(301),
+            TestUtils.createBlockHeaderMock(303)
+        );
+
+        List<Block> testBlocks = Arrays.asList(
+            new Block(block1Header, Collections.emptyList(), block1Brothers, true, true),
+            new Block(block2Header, Collections.emptyList(), block2Brothers, true, true)
+        );
+
+        AdvanceBlockchainMessage message = new AdvanceBlockchainMessage(testBlocks);
+        List<String> parsedBlockHeaders = message.getParsedBlockHeaders();
+
+        String[] blockBrothers1 = message.getParsedBrothers(parsedBlockHeaders.get(1));
+        for (int i = 0; i < blockBrothers1.length - 1; ++i) {
```

Comment:
Any particular reason to use ++i instead of i++? I think it's prettu much the same but I'm simply used to seeing i++

Reply:
Updated